### PR TITLE
E2E network management

### LINKED ISSUE #5083: Add e2e test for network management
### Acceptance Criteria

Pre-requisite: 

- Ensure you have two accounts added to the wallet and one account has been selected and we are in `Wallet` landing page.
- Ensure you have more than one network supported
- Ensure you have more than one application for switching

1. Select available network
2. Add a new network
3. Switch between available application
4. Add available application from the list

### PR DESCRIPTION
**Pull request by oskarleonard**
### What was the problem?

This PR resolves #5083

### How was it solved?

Added E2E tests for
1. Adding network
2. Switch network
3. Edit network
4. Delete network

Also removed unnecessary logic in before hook since we now can use a stable local network (customNode) when running the tests in jenkins.

### How was it tested?

1. git pull
2. `DEFAULT_NETWORK=customNode yarn dev`
3. `PW_BASE_URL=http://localhost:8080/# yarn run cucumber:playwright:open`
4. Expected: Tests should pass

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -37,13 +37,13 @@ pipeline {
 		stage('build') {
 			steps {
 				nvm(getNodejsVersion()) {
-					withEnv(["DEFAULT_NETWORK=testnet"]) {
+					withEnv(["DEFAULT_NETWORK=customNode"]) {
```
Comment (eniolam1000752) [inline Jenkinsfile]: we should not do this because any user who uses the route from the deployment the default network they would access would be `customNetwork` which would not be available for them.

File: e2e/features/networkManagement/NetworkManagement.feature [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  Scenario: Error message when adding an existing network
+    Given I click on text "Add network"
+    Then I should be redirected to route: 'wallet?modal=dialogAddNetwork'
+    And button with text "Add network" should be disabled
+    Given I type "custom_devnet" in "name"
+    And I type "http://devnet-service.liskdev.net:9901" in "serviceUrl"
+    And I click on a button with text "Add network"
+    Then I should see "Name & ServiceUrl already exists."
+
+  Scenario: Error message add a wrong serviceUrl
```
Comment (eniolam1000752) [inline e2e/features/networkManagement/NetworkManagement.feature:45]:
```suggestion
  Scenario: Error message when a wrong serviceUrl is inputted
```

File: e2e/features/networkManagement/NetworkManagement.feature [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    Given I click on text "Add network"
+    Then I should be redirected to route: 'wallet?modal=dialogAddNetwork'
+    And button with text "Add network" should be disabled
+    Given I type "custom_devnet" in "name"
+    And I type "http://devnet-service.liskdev.net:9901" in "serviceUrl"
+    And I wait for "1 seconds"
+    And I click on a button with text "Add network"
+    Then I should see 'Custom network added "custom_devnet"'
+
+  Scenario: Edit network
```
Comment (eniolam1000752) [inline e2e/features/networkManagement/NetworkManagement.feature]:
```suggestion
  Scenario: Edit custom network
```

File: src/modules/network/utils/getNetwork.test.js [hunk 4]
```diff
@@ -9,7 +9,7 @@ describe('Utils: getNetwork', () => {
       { label: 'Betanet', name: 'betanet' },
       { label: 'Devnet', name: 'devnet' },
       { label: 'Testnet', name: 'testnet' },
-      { label: 'Custom Service Node', name: 'customNode' },
+      { label: 'customNode', name: 'customNode' },
```
Comment (eniolam1000752) [inline src/modules/network/utils/getNetwork.test.js]:
```suggestion
      { label: 'CustomNode', name: 'customNode' },
```

for consistency

File: e2e/features/networkManagement/NetworkManagement.feature [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  Background: Add an account and navigate open Switch network modal
+    Given I go to page "wallet"
+    And I wait for "1 seconds"
+    And I click on a button with text "customNode"
+    Then "customNode" should be selected in "Switch network" dropdown
+    Given I click on text "Add network"
+    Then I should be redirected to route: 'wallet?modal=dialogAddNetwork'
+    And button with text "Add network" should be disabled
+    Given I type "custom_devnet" in "name"
+    And I type "http://devnet-service.liskdev.net:9901" in "serviceUrl"
```
Comment (eniolam1000752) [inline e2e/features/networkManagement/NetworkManagement.feature:11]: I think we should also have a step that covers for when a user inputs ws url.

**Review by eniolam1000752 — CHANGES_REQUESTED** [review]

File: e2e/features/networkManagement/NetworkManagement.feature [hunk 6]
```diff
@@ -0,0 +1,54 @@
+Feature: Network Management
+  Background: Add an account and navigate open Switch network modal
+    Given I go to page "wallet"
+    And I wait for "1 seconds"
+    And I click on a button with text "customNode"
+    Then "customNode" should be selected in "Switch network" dropdown
+    Given I click on text "Add network"
+    Then I should be redirected to route: 'wallet?modal=dialogAddNetwork'
```
Comment (ikem-legend) [inline e2e/features/networkManagement/NetworkManagement.feature]: I feel this should be "Then I should be redirected to addNetwork modal" or something similar. That way other modal views will be written in the same manner and the step definition can use the routes file

File: e2e/features/networkManagement/NetworkManagement.feature (hunk 3 above)
Comment (ikem-legend) [reply e2e/features/networkManagement/NetworkManagement.feature]: @eniolam1000752 If only custom networks can be edited, is it necessary to rename this is as "edit custom network"?

File: e2e/features/networkManagement/NetworkManagement.feature (hunk 6 above)
Comment (oskarleonard) [reply e2e/features/networkManagement/NetworkManagement.feature]: Fixed https://github.com/LiskHQ/lisk-desktop/pull/5345/commits/48b2b682db531c4a62f09551ec22c8e497f293cd

File: e2e/features/networkManagement/NetworkManagement.feature (hunk 5 above)
Comment (oskarleonard) [reply e2e/features/networkManagement/NetworkManagement.feature:11]: Added it to background

File: e2e/features/networkManagement/NetworkManagement.feature (hunk 3 above)
Comment (oskarleonard) [reply e2e/features/networkManagement/NetworkManagement.feature]: Agree changed to more Correct, Edit network name

File: e2e/features/account/AddAccount.feature [hunk 7]
```diff
@@ -30,102 +30,5 @@ Feature: AddAccount
     And I type "Password1$" in "cPassword"
     And I type "test_acc" in "accountName"
     And I click on text "I agree to store my encrypted secret recovery phrase on this device."
-    And I click on a button with text "Save Account"
+    And I click on exact text "Save Account"
     Then I should see the final add account step
-
```
Comment (ikem-legend) [inline e2e/features/account/AddAccount.feature]: Why were these removed?

Comment (oskarleonard) [reply e2e/features/account/AddAccount.feature]: Goot catch, must have removed them during a merge conflict

**Review by ikem-legend — APPROVED** [review]

**Review by ManuGowda — APPROVED** [review]
